src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/vilkårperiode/domain/FaktaOgVurderingMapper.kt

```diff
@@ -53,10 +54,20 @@ import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.dto.LagreVilkårperiod
 fun mapFaktaOgSvarDto(
     vilkårperiode: LagreVilkårperiode,
     stønadstype: Stønadstype,
+    vurderingAldersvilkår: VurderingAldersVilkår?,
 ): FaktaOgVurdering =
     when (vilkårperiode.type) {
         is AktivitetType -> mapAktiviteter(stønadstype = stønadstype, aktivitet = vilkårperiode)
-        is MålgruppeType -> mapMålgruppe(stønadstype = stønadstype, målgruppe = vilkårperiode)
+        is MålgruppeType ->
+            mapMålgruppe(
+                stønadstype = stønadstype,
+                målgruppe = vilkårperiode,
+                vurderingAldersvilkår =
+                    vurderingAldersvilkår.let {
+                        it
+                            ?: error("Mangler vurdering av aldersvilkår for målgruppe=$vilkårperiode, stønadstype=$stønadstype")
```

Comment:
Er det mapper-funksjonen sin jobb å kaste denne feilmeldingen? Jeg føler det burde skje på et tidligere stadie kanskje.

Skjønner heller ikke helt logikken her. Slik jeg leser denne koden vil aldersvilkåret være obligatorisk for alle målgrupper, men slik er det vel ikke?

Reply:
Mapper kaster ikke feil slik koden er nå